src/wasm-wasi-component/src/lib.rs

```diff
@@ -209,11 +210,15 @@ impl GlobalState {
         let component = Component::from_file(&engine, &global_config.component)
             .context("failed to compile component")?;
         let mut linker = Linker::<StoreState>::new(&engine);
-        wasmtime_wasi::preview2::command::add_to_linker(&mut linker)?;
-        wasmtime_wasi_http::proxy::add_only_http_to_linker(&mut linker)?;
+        add_to_linker_async(&mut linker)
+            .context("failed to add to linker")?;
```

Comment:
Is this what we want?

We're going from adding all of `wasi:cli/command` and `wasi:http/proxy` to only adding `wasi:http/proxy`.

We probably want to add in at least:

```rust
        wasmtime_wasi::bindings::cli::environment::add_to_linker(&mut linker)
            .context("failed to add wasi:cli/environment to linker")?;
```

...and perhaps others.

Reply:
Trying to build with this patch

```diff
diff --git ./src/wasm-wasi-component/src/lib.rs ./src/wasm-wasi-component/src/lib.rs
index faff2cc8..c6cb91f9 100644
--- ./src/wasm-wasi-component/src/lib.rs
+++ ./src/wasm-wasi-component/src/lib.rs
@@ -210,6 +210,8 @@ impl GlobalState {
         let component = Component::from_file(&engine, &global_config.component)
             .context("failed to compile component")?;
         let mut linker = Linker::<StoreState>::new(&engine);
+        wasmtime_wasi::bindings::cli::environment::add_to_linker(&mut linker)
+            .context("failed to add wasi:cli/environment to linker")?;
         add_to_linker_async(&mut linker)
             .context("failed to add to linker")?;
         //add_only_http_to_linker_sync(&mut linker)
```

Reply:
Heh

```
error[E0061]: this function takes 2 arguments but 1 argument was supplied
   --> src/lib.rs:213:9
    |
213 | ...   wasmtime_wasi::bindings::cli::environment::add_to_linker(&mut linker)
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^------------- an argument is missing
    |
note: function defined here
   --> /home/andrew/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasmtime-wasi-24.0.0/src/bindings.rs:324:5
    |
324 | /     wasmtime::component::bindgen!({
325 | |         path: "wit",
326 | |         world: "wasi:cli/command",
327 | |         tracing: true,
...   |
415 | |         },
416 | |     });
    | |______^
    = note: this error originates in the macro `wasmtime::component::bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
    |
213 |         wasmtime_wasi::bindings::cli::environment::add_to_linker(&mut linker, /* get */)
    |                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `wasm-wasi-component` (lib) due to 1 previous error
make: *** [build/Makefile:2150: src/wasm-wasi-component/target/release/libwasm_wasi_component.so] Error 101
```

I'll poke it some more after food...

Reply:
Oh of course. For now, let's just throw the whole kitchen sink in there

    wasmtime_wasi::add_to_linker_async(&mut linker)

Then confirm that the failing component builds and runs. It _won't_ see envvars, since the ones we want to propagate need to get explicitly added by calling [`envs`](https://docs.rs/wasmtime-wasi/24.0.0/wasmtime_wasi/struct.WasiCtxBuilder.html#method.envs), [`env`](https://docs.rs/wasmtime-wasi/24.0.0/wasmtime_wasi/struct.WasiCtxBuilder.html#method.env), or [`inherit_env`](https://docs.rs/wasmtime-wasi/24.0.0/wasmtime_wasi/struct.WasiCtxBuilder.html#method.inherit_env) on the WasiCtxBuilder

Reply:
Setting a top-level review comment; marking this thread as resolved.